src/trans/gpu/external/inv_transad.F90

```diff
@@ -160,7 +171,463 @@ SUBROUTINE INV_TRANSAD(PSPVOR,PSPDIV,PSPSCALAR,PSPSC3A,PSPSC3B,PSPSC2,&
+    ENDIF
+    IF(IUBOUND(3) < NGPBLKS) THEN
+      WRITE(NOUT,*)'INV_TRANSAD:THIRD DIM. OF PGP2 TOO SMALL/LARGE ',IUBOUND(3),NGPBLKS
+      CALL ABORT_TRANS('INV_TRANSAD:THIRD DIMENSION OF PGP2 TOO SMALL/LARGE ')
+    ENDIF
+  ELSE
+    CALL ABORT_TRANS('INV_TRANSAD:PGP2 MISSING')
+  ENDIF
+ENDIF
+
+IF(PRESENT(PGP3A)) THEN
+  IF(.NOT.PRESENT(PSPSC3A)) THEN
+    CALL ABORT_TRANS('INV_TRANSAD:PSPSC3A HAS TO BE PRESENT WHEN PGP3A IS')
+  ENDIF
+ENDIF
+IF(IF_SC3A_G3 > 0) THEN
+  IF(PRESENT(PGP3A)) THEN
+    IUBOUND=UBOUND(PGP3A)
+    IF(IUBOUND(1) < NPROMA) THEN
+      WRITE(NOUT,*)'INV_TRANSAD:FIRST DIM. OF PGP3A TOO SMALL/LARGE ',IUBOUND(1),NPROMA
+      CALL ABORT_TRANS('INV_TRANSAD:FIRST DIMENSION OF PGP3A TOO SMALL/LARGE ')
+    ENDIF
+    IF(IUBOUND(2) /= IF_SC3A_G2) THEN
+      WRITE(NOUT,*)'INV_TRANSAD:SEC DIM. OF PGP3A INCONSISTENT ',IUBOUND(2),IF_SC3A_G2
+      CALL ABORT_TRANS('INV_TRANSAD:SEC. DIMENSION OF PGP3A INCONSISTENT ')
+    ENDIF
+    IF(IUBOUND(3) /= IF_SC3A_G3 ) THEN
+      WRITE(NOUT,*)'INV_TRANSAD:THIRD DIM. OF PGP3A INCONSISTENT ',&
+       & IUBOUND(3),IF_SC3A_G3
+      CALL ABORT_TRANS('INV_TRANSAD:THIRD DIMENSION OF PGP3A INCONSISTENT ')
+    ENDIF
+    IF(IUBOUND(4) < NGPBLKS) THEN
+      WRITE(NOUT,*)'INV_TRANSAD:FOURTH DIM. OF PGP3A TOO SMALL/LARGE ',IUBOUND(4),NGPBLKS
+      CALL ABORT_TRANS('INV_TRANSAD:FOURTH DIMENSION OF PGP3A TOO SMALL/LARGE ')
+    ENDIF
+  ELSE
+    CALL ABORT_TRANS('INV_TRANSAD:PGP3A MISSING')
+  ENDIF
+ENDIF
+
+IF(PRESENT(PGP3B)) THEN
+  IF(.NOT.PRESENT(PSPSC3B)) THEN
+    CALL ABORT_TRANS('INV_TRANSAD:PSPSC3B HAS TO BE PRESENT WHEN PGP3B IS')
+  ENDIF
+ENDIF
+IF(IF_SC3B_G3 > 0) THEN
+  IF(PRESENT(PGP3B)) THEN
+    IUBOUND=UBOUND(PGP3B)
+    IF(IUBOUND(1) < NPROMA) THEN
+      WRITE(NOUT,*)'INV_TRANSAD:FIRST DIM. OF PGP3B TOO SMALL/LARGE ',IUBOUND(1),NPROMA
+      CALL ABORT_TRANS('INV_TRANSAD:FIRST DIMENSION OF PGP3B TOO SMALL/LARGE ')
+    ENDIF
+    IF(IUBOUND(2) /= IF_SC3B_G2) THEN
+      WRITE(NOUT,*)'INV_TRANSAD:SEC DIM. OF PGP3B INCONSISTENT ',IUBOUND(2),IF_SC3B_G2
+      CALL ABORT_TRANS('INV_TRANSAD:SEC. DIMENSION OF PGP3B INCONSISTENT ')
+    ENDIF
+    IF(IUBOUND(3) /= IF_SC3B_G3 ) THEN
+      WRITE(NOUT,*)'INV_TRANSAD:THIRD DIM. OF PGP3B INCONSISTENT ',&
+       & IUBOUND(3),IF_SC3B_G3
+      CALL ABORT_TRANS('INV_TRANSAD:THIRD DIMENSION OF PGP3B INCONSISTENT ')
+    ENDIF
+    IF(IUBOUND(1) < NPROMA) THEN
+      WRITE(NOUT,*)'INV_TRANSAD:FIRST DIM. OF PGP3B TOO SMALL/LARGE ',IUBOUND(1),NPROMA
+      CALL ABORT_TRANS('INV_TRANSAD:FIRST DIMENSION OF PGP3B TOO SMALL/LARGE ')
+    ENDIF
+  ELSE
+    CALL ABORT_TRANS('INV_TRANSAD:PGP3B MISSING')
+  ENDIF
+ENDIF
+CALL GSTATS(1807,1)
+
+!     ------------------------------------------------------------------
+
+! Perform transform
+
+CALL INV_TRANS_CTLAD(IF_UV_G,IF_SCALARS_G,IF_GP,IF_FS,IF_OUT_LT,&
+ & IF_UV,IF_SCALARS,IF_SCDERS,&
+ & PSPVOR,PSPDIV,PSPSCALAR,KVSETUV,KVSETSC,PGP,FSPGL_PROC,&
```

Comment:
```suggestion
 & PSPVOR,PSPDIV,PSPSCALAR,KVSETUV,KVSETSC,PGP,&
```

`FSPGL_PROC` isn't used by `INV_TRANSAD` (even though it is an argument) so we can delete this.